src/services/splitHttpClient.ts

```diff
@@ -1,22 +1,21 @@
 import { IFetch, IRequestOptions, ISplitHttpClient } from './types';
 import { SplitError, SplitNetworkError } from '../utils/lang/errors';
 import objectAssign from 'object-assign';
-import { IMetadata } from '../dtos/types';
-import { ILogger } from '../logger/types';
 import { ERROR_HTTP, ERROR_CLIENT_CANNOT_GET_READY } from '../logger/constants';
+import { ISettings } from '../types';
 
 const messageNoFetch = 'Global fetch API is not available.';
 
 /**
  * Factory of Split HTTP clients, which are HTTP clients with predefined headers for Split endpoints.
  *
- * @param apikey api key to set Authorization header
- * @param metadata data to set additional headers
+ * @param settings SDK settings, used to access authorizationKey, logger instance and metadata (SDK version, ip and hostname) to set additional headers
  * @param options global request options
  * @param fetch optional http client to use instead of the global Fetch (for environments where Fetch API is not available such as Node)
  */
-export function splitHttpClientFactory(log: ILogger, apikey: string, metadata: IMetadata, getFetch?: () => (IFetch | undefined), getOptions?: () => object): ISplitHttpClient {
+export function splitHttpClientFactory(settings: Pick<ISettings, 'log' | 'version' | 'runtime' | 'core'>, getFetch?: () => (IFetch | undefined), getOptions?: () => object): ISplitHttpClient {
```

Comment:
Same here: when function is more than 2 parameters, use new lines for each of them.